dbt/models/reporting/reporting.ratio_stats.py

```diff
@@ -1,157 +1,129 @@
+    ratio_not_outlier = ratio.between(
+        ratio.quantile(CCAO_LOWER_QUANTILE),
+        ratio.quantile(CCAO_UPPER_QUANTILE),
+        inclusive="neither",
+    ).reset_index(drop=True)
 
-def ccao_cod(ratio):
-    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
-    no_outliers = ratio[
-        ratio.between(
-            ratio.quantile(0.05), ratio.quantile(0.95), inclusive="neither"
-        )
-    ]
+    estimate_no_outliers = estimate[ratio_not_outlier]
+    sale_price_no_outliers = sale_price[ratio_not_outlier]
+    n: float = float(estimate_no_outliers.size)
 
-    cod_n = no_outliers.size
+    return estimate_no_outliers, sale_price_no_outliers, n
 
-    if cod_n >= 20:
-        cod_val = cod(no_outliers)
-        cod_ci = cod_boot(no_outliers, nboot=1000)
-        cod_ci = f"{cod_ci[0]}, {cod_ci[1]}"
-        met = cod_met(cod_val)
-        out = [cod_val, cod_ci, met, cod_n]
 
+def ccao_metric(
+    fun: callable,
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+) -> dict:
+    """
+    Helper function to calculate a metric, its confidence interval, and
+    whether the metric meets the IAAO/Quintos standard. Also checks if the
+    number of samples is large enough to calculate a metric. Per CCAO
+    standards, the sample size must be at least 20.
+    """
+    est_no_out, sale_no_out, n = ccao_drop_outliers(estimate, sale_price)
+
+    if n >= CCAO_MIN_SAMPLE_SIZE:
+        val = getattr(ap, fun)(est_no_out, sale_no_out)
+        # MKI doesn't have a _ci function, so we need a check here to return
+        # None if it is called
+        try:
+            ci_l, ci_u = getattr(ap, f"{fun}_ci")(
+                est_no_out, sale_no_out, nboot=300
+            )
+        except AttributeError:
+            ci_l, ci_u = None, None
+        met = getattr(ap, f"{fun}_met")(val)
+        out = [val, ci_l, ci_u, met, n]
     else:
-        out = [None, None, None, cod_n]
-
-    return out
-
+        out = [None, None, None, None, n]
 
-def ccao_prd(fmv, sale_price):
-    ratio = fmv / sale_price
-
-    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
-    no_outliers = ratio.between(
-        ratio.quantile(0.05), ratio.quantile(0.95), inclusive="neither"
+    # Zip into a dictionary for use with the calc_summary function below,
+    # which expects a dict to expand into a DataFrame
+    out_dict = dict(
+        zip([fun, f"{fun}_ci_l", f"{fun}_ci_u", f"{fun}_met", f"{fun}_n"], out)
     )
 
-    fmv_no_outliers = fmv[no_outliers]
-    sale_price_no_outliers = sale_price[no_outliers]
+    return out_dict
 
-    prd_n = sum(no_outliers)
 
-    if prd_n >= 20:
-        prd_val = prd(fmv_no_outliers, sale_price_no_outliers)
-        prd_ci = prd_boot(fmv_no_outliers, sale_price_no_outliers, nboot=1000)
-        prd_ci = f"{prd_ci[0]}, {prd_ci[1]}"
-        met = prd_met(prd_val)
+def ccao_median(
```

Comment:
Median gets its own function because it's not included as a stat in `assesspy` and takes different inputs (`ratio`, as opposed to `estimate` and `sale_price`).